### PR TITLE
Update force merge polling

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Change force merge polling logic to use wait_for_completion
to false, to make it async and use task's get api to check
whether task is completed or not to exit from force merge.

Here, request end time is calcuated only after task is completed.
This request end time will not be 100% accurate, since, we use polling to check
whether task status is completed or not.

### Testing
- [x] New functionality includes testing

[Describe how this change was tested]

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VijayanB** [comment]
This depends on #488

**Review by navneet1v — APPROVED** [review]

**Comment by VijayanB** [comment]
@gkamat @IanHoang Can this be reviewed and added to 1.4 release?

**Comment by VijayanB** [comment]
@gkamat @IanHoang Any update on this?

File: osbenchmark/worker_coordinator/runner.py
Comment (gkamat) [inline osbenchmark/worker_coordinator/runner.py:686]: Please return a diagnostic if `wait_for_completion` is provided in params and happens to be set to `true`.  This will revert to the blocking mode in any case.

File: osbenchmark/worker_coordinator/runner.py
Comment (gkamat) [inline osbenchmark/worker_coordinator/runner.py:691]: Do not fall into an infinite loop if the `get` returns no task for some reason and provide an exit.

File: osbenchmark/worker_coordinator/runner.py
Comment (gkamat) [inline osbenchmark/worker_coordinator/runner.py:693]: The task document should be deleted once done with.  " When you are done with a task, you should delete the task document so ... can reclaim the space."  It's generally not an issue, but good practice in case workloads are run in a loop for longevity tests.

File: tests/worker_coordinator/runner_test.py
Comment (gkamat) [inline tests/worker_coordinator/runner_test.py:1186]: Perhaps use a non-zero poll period?

File: osbenchmark/worker_coordinator/runner.py
Comment (gkamat) [inline osbenchmark/worker_coordinator/runner.py:694]: What happens when a connection timeout is thrown if a long poll period is specified and control returns to the caller?

File: tests/worker_coordinator/runner_test.py
Comment (gkamat) [inline tests/worker_coordinator/runner_test.py:1167]: Should there be a test where a connection timeout is thrown when a long poll period is in effect?

**Review by gkamat — CHANGES_REQUESTED** [review]

File: osbenchmark/worker_coordinator/runner.py
Comment (VijayanB) [reply osbenchmark/worker_coordinator/runner.py:693]: I thought about it. I don't see an api to remove task document directly. Do you aware of an api for that? I don't want to delete document using delete document api from .tasks index.

File: tests/worker_coordinator/runner_test.py
Comment (VijayanB) [reply tests/worker_coordinator/runner_test.py:1186]: Ack

File: tests/worker_coordinator/runner_test.py
Comment (VijayanB) [reply tests/worker_coordinator/runner_test.py:1167]: This will be applicable to every api. i don't see connection timeout was specifically called out. In previous case, i understand it could be a problem due to synchronous task. I don't think it is problem in this case. However, if you have strong feeling i can write another test that raise connection timeout error.

File: osbenchmark/worker_coordinator/runner.py
Comment (VijayanB) [reply osbenchmark/worker_coordinator/runner.py:691]: Ack

File: osbenchmark/worker_coordinator/runner.py
Comment (VijayanB) [reply osbenchmark/worker_coordinator/runner.py:686]: @gkamat Sorry i don't understand what do you mean return a diagnostic. Can you provide an example on how to do that?

File: osbenchmark/worker_coordinator/runner.py
Comment (VijayanB) [reply osbenchmark/worker_coordinator/runner.py:694]: Did you mean when connection timeout is thrown during get task api?

File: osbenchmark/worker_coordinator/runner.py
Comment (gkamat) [reply osbenchmark/worker_coordinator/runner.py:692]: You can print a warning message using `console.warn()`.

File: osbenchmark/worker_coordinator/runner.py
Comment (gkamat) [reply osbenchmark/worker_coordinator/runner.py:710]: You are correct.  This is a transient task, so ideally, the record should be removed explicitly; however the behavior is different between Elasticsearch and OpenSearch.  Perhaps this will change in the future, so ignore this for now.

File: osbenchmark/worker_coordinator/runner.py
Comment (VijayanB) [reply osbenchmark/worker_coordinator/runner.py:692]: Ack

File: osbenchmark/worker_coordinator/runner.py
Comment (gkamat) [reply osbenchmark/worker_coordinator/runner.py:716]: Yes, admittedly a small possibility, but any exceptions raised by the client will get caught by the caller otherwise.

File: tests/worker_coordinator/runner_test.py
Comment (gkamat) [reply tests/worker_coordinator/runner_test.py:1167]: Not a major issue; I will leave it up to you.  It was just a suggestion.

**Review by gkamat — CHANGES_REQUESTED** [review]
Just responded to your earlier comments.  Will take a look at the new updates that were just force-pushed when I get a chance, for approval.  Thanks for the changes.

File: osbenchmark/worker_coordinator/runner.py
Comment (IanHoang) [inline osbenchmark/worker_coordinator/runner.py:694]: Will there ever be a case where `mode == "polling"` but `self.PARAM_WAIT_FOR_COMPLETION` is not in `params`? 

Even if `self.PARAM_WAIT_FOR_COMPLETION` is not found in params, `merge_params` will always set its key of `wait_for_completion` to be "false" anyways. If this is the case, we can remove this if statement and just have the logging statement aligned with the rest.

```
# Original 
        if mode == "polling":
            if params.get(self.PARAM_WAIT_FOR_COMPLETION):
                self.logger.warning(
                    "%s is set for polling. It will be updated to false", self.PARAM_WAIT_FOR_COMPLETION)
            merge_params[self.PARAM_WAIT_FOR_COMPLETION] = "false"

# New
       if mode == "polling":
                self.logger.warning(
                    "%s is set for polling. It will be updated to false", self.PARAM_WAIT_FOR_COMPLETION)
                merge_params[self.PARAM_WAIT_FOR_COMPLETION] = "false"
```

File: osbenchmark/worker_coordinator/runner.py
Comment (IanHoang) [reply osbenchmark/worker_coordinator/runner.py:693]: That way we document this disclaimer for now and we can also include it into the documentation when that's ready

**Review by IanHoang — CHANGES_REQUESTED** [review]
Left a couple comments

File: osbenchmark/worker_coordinator/runner.py
Comment (VijayanB) [reply osbenchmark/worker_coordinator/runner.py:693]: Ack

File: osbenchmark/worker_coordinator/runner.py
Comment (VijayanB) [reply osbenchmark/worker_coordinator/runner.py:694]: Ack

**Comment by VijayanB** [comment]
@IanHoang Updated PR with your feedback, Can you take a look one more time ? Thanks.

**Review by IanHoang — APPROVED** [review]

**Comment by VijayanB** [comment]
@gkamat Can you take a look at this PR? This change is blocker for some of our runs that depends on large force merge time.

**Review by gkamat — APPROVED** [review]

**Pull request merged**
